### PR TITLE
mini-hlint : add mini-hlint example

### PR DESCRIPTION
**Pull request by shayne-fletcher-da**
This PR adds a small example. Attempt to find "double negations" in a module.

### CODE REVIEW

File: examples/mini-hlint/mini-hlint.cabal [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+cabal-version:       >=1.10
+
+executable mini-hlint
+  main-is:             Main.hs
+  build-depends:       base >=4.11
+                     , extra
+                     , ghc-lib
+                     , time
+  default-language:    Haskell2010
+  default-extensions:  PackageImports
```
Comment (neil-da) [inline examples/mini-hlint/mini-hlint.cabal]: I'd suggest putting this in the file rather than here.

File: examples/mini-hlint/mini-hlint.cabal [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  ghc-options:         -Wno-missing-fields
```
Comment (neil-da) [inline examples/mini-hlint/mini-hlint.cabal]: Ditto

File: examples/mini-hlint/mini-hlint.cabal [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+copyright:           Digital Asset 2018-2019
+build-type:          Simple
+cabal-version:       >=1.10
+
+executable mini-hlint
+  main-is:             Main.hs
+  build-depends:       base >=4.11
+                     , extra
+                     , ghc-lib
+                     , time
```
Comment (neil-da) [inline examples/mini-hlint/mini-hlint.cabal]: Is time required?

File: examples/mini-hlint/src/Main.hs [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+import "ghc-lib" ErrUtils
+import qualified "ghc-lib" Parser
+
+import FastString
+import Outputable
+import SrcLoc
+import System.Environment
+import System.IO.Extra
+import Control.Monad
+import FastString
```
Comment (neil-da) [inline examples/mini-hlint/src/Main.hs]: Dupe

File: examples/mini-hlint/src/Main.hs [hunk 5]
```diff
@@ -0,0 +1,103 @@
+module Main where
```
Comment (neil-da) [inline examples/mini-hlint/src/Main.hs]:
```suggestion
module Main(main) where
```

File: examples/mini-hlint/src/Main.hs (hunk 4 above)
Comment (neil-da) [reply examples/mini-hlint/src/Main.hs]: HLint would have caught this

File: examples/mini-hlint/test/MiniHlintTest.hs [hunk 6]
```diff
@@ -0,0 +1,4 @@
+module MiniHlintTest where
+
+f :: Bool -> Int
+f x = 1 + if id (not (not x)) || false then 1 else 2
```
Comment (neil-da) [inline examples/mini-hlint/test/MiniHlintTest.hs]:
```suggestion
f x = 1 + if id (not (not x)) || False then 1 else 2
```

File: examples/mini-hlint/mini-hlint.cabal (hunk 1 above)
Comment (shayne-fletcher-da) [reply examples/mini-hlint/mini-hlint.cabal]: Ok

File: examples/mini-hlint/mini-hlint.cabal (hunk 2 above)
Comment (shayne-fletcher-da) [reply examples/mini-hlint/mini-hlint.cabal]: Ok

File: examples/mini-hlint/mini-hlint.cabal (hunk 3 above)
Comment (shayne-fletcher-da) [reply examples/mini-hlint/mini-hlint.cabal]: No

File: examples/mini-hlint/src/Main.hs (hunk 4 above)
Comment (shayne-fletcher-da) [reply examples/mini-hlint/src/Main.hs]: Fixed `No hints`.

File: examples/mini-hlint/src/Main.hs (hunk 5 above)
Comment (shayne-fletcher-da) [reply examples/mini-hlint/src/Main.hs]: Ok

File: examples/mini-hlint/test/MiniHlintTest.hs (hunk 6 above)
Comment (shayne-fletcher-da) [reply examples/mini-hlint/test/MiniHlintTest.hs]: Oops!

**Pull request closed without merging**